canonical-web-and-design/maas-ui
# refactor(ui): convert PowerTypeFields to TS and move to base components

## Done

- Converted `PowerTypeFields` to typescript and connected it to the store so it retrieves the power types itself instead of need to be given them as a prop
- Fixed some iffy logic around field scopes and chassis power types. Now they're two entirely separate props (`forChassis` and `fieldScopes`) so it's up to each particular form to determine which subset of power fields to show. This will be necessary for the machine power form.
- Added some enums

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that the forms for Add Machine, Add Chassis and Add KVM all still validate properly
- Check that you can still add a manual machine.

First part of canonical-web-and-design/MAAS-squad#2333
